client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js

```diff
@@ -1,170 +1,205 @@
+      this.$uibModal = $uibModal;
+      this.OvhApiTelephony = OvhApiTelephony;
+      this.TucToast = TucToast;
+      this.TelephonyMediator = TelephonyMediator;
     }
 
-    this.agents.paginated = _.sortBy(this.agents.paginated, prop);
-
-    if (!this.orderedAscAgents) {
-      this.agents.paginated = this.agents.paginated.reverse();
-    }
-  }
-
-  fetchAgentsIds() {
-    this.agents.isLoading = true;
-    return this.OvhApiTelephony.OvhPabx().Hunting().Agent().v6()
-      .query({
-        billingAccount: this.$stateParams.billingAccount,
-        serviceName: this.$stateParams.serviceName,
-      }).$promise.finally(() => {
-        this.agents.isLoading = false;
-      });
-  }
-
-  fetchAgent(id) {
-    return this.OvhApiTelephony.OvhPabx().Hunting().Agent().v6()
-      .get({
-        billingAccount: this.$stateParams.billingAccount,
-        serviceName: this.$stateParams.serviceName,
-        agentId: id,
-      }).$promise;
-  }
-
-  getSelectedAgentIds() {
-    return _.keys(this.agents.selected);
-  }
-
-  onTransformItemDone(items) {
-    this.$timeout(() => this.orderBy('number', this.orderedAscAgents));
-    return items;
-  }
-
-  deleteAgents() {
-    this.agents.isDeleting = true;
-    return this.$q
-      .all(this.getSelectedAgentIds()
-        .map(id => this.OvhApiTelephony.OvhPabx().Hunting().Agent().v6()
-          .remove({
-            billingAccount: this.$stateParams.billingAccount,
-            serviceName: this.$stateParams.serviceName,
-            agentId: id,
-          }).$promise.then(() => {
-            _.pull(this.agents.id, parseInt(id, 10));
-          })))
-      .then(() => {
-        this.TucToast.success(this.$translate.instant('telephony_alias_configuration_agents_delete_success'));
-        return this.fetchAgentsIds().then((ids) => {
-          this.agents.ids = ids;
-        });
-      })
-      .catch((err) => {
-        this.TucToast.error([this.$translate.instant('telephony_alias_configuration_agents_delete_error'), _.get(err, 'data.message')].join(' '));
-        return this.$q.reject(err);
-      })
-      .finally(() => {
-        this.agents.isDeleting = false;
-        this.agents.selected = {};
+    $onInit() {
+      this.orderedAscAgents = true;
+      this.agents = {
+        ids: [],
+        paginated: [],
+        selected: {},
+        isLoading: false,
+      };
+
+      this.addAgentForm = {
+        numbers: [null],
+        isAdding: false,
+      };
+
+      this.types = [
+        { value: 'internal', label: this.$translate.instant('telephony_alias_configuration_agents_type_internal') },
+        { value: 'external', label: this.$translate.instant('telephony_alias_configuration_agents_type_external') },
+      ];
+
+      this.onChooseServicePopover = ({ serviceName }, pos) => {
+        this.addAgentForm.numbers[pos] = serviceName;
+      };
+
+      this.$q.all({
+        translations: this.$translate.refresh(),
+        services: this.TelephonyMediator.getAll(),
```

Comment:
You can use this service : https://github.com/ovh-ux/telecom-universe-components/blob/master/src/telecom/voip/telecom-voip.service.js#L52 to get all services.

For a better explanation, `TelephonyMediator` is meant to be replaced completely by cleaner service `voipService`.
(Also, just know that`voipService` and its children classes don't cover all existing features of TelephonyMediator))

Reply:
`this.tucTelecomVoip.fetchAll()` returns you all billingAccounts with services in.
So `TelephonyMediator.groups` can be replaced by `result.services` (Should be rename as `result.billingAccounts` for clarity)